.github/actions/pgx-init/action.yml

```diff
@@ -0,0 +1,41 @@
+name: 'pgx initialization'
+description: 'cargo pgx init'
+inputs:
+  working-directory:
+    description: 'The directory in which there is a pgx extension project'
+    required: true
+  pg_version:
+    description: 'The version of postgres to use'
+    required: false
+    default: 14
+outputs: {}
+runs:
+  using: "composite"
+  steps:
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: ${{ inputs.working-directory }}
+          workspaces: |
+            ${{ inputs.working-directory }}
+          # Additional directories to cache
+          cache-directories: |
+            /home/runner/.pgx
+      - name: setup pgx
+        shell: bash
+        run: cargo install --locked cargo-pgx
+      - name: pgx init
+        shell: bash
+        run: |
+          set -x
+          ls -ltrah /home/runner/.pgx || true
+          # pgx init can take a long time, and it re-compiles postgres even when there
+          # is a cached version. So, we can just check for the directory and
+          if find /home/runner/.pgx | grep $(awk -F "=" '/pg$${{ inputs.pg_version }}/ {print $2}' /home/runner/.pgx/config.toml | tr -d '"'); then
+            echo "Already found pgx is initialized. Skipping 'cargo pgx init' command."
+          else
+            cargo pgx init --pg${{ inputs.pg_version }} download
+          fi
```

Comment:
The logic here is to avoid running pgx init on a cache hit. The reasoning of the logic is here in the pgx discord https://discordapp.com/channels/561648697805504526/749454413407584258/1055239589276700795